### ISSUE TITLE
Suggest image descriptions (alt text) in case they are missing

### ISSUE THREAD

**Issue by OddDev** [mastodon/mastodon#26464]
### Pitch

Für posts that feature media without a11y relevant descriptions add a button to "suggest alt text". One can add a suitable description for the author to approve. 

### Motivation

People that are not able to fully visually comprehend an image rely on image descriptions. Often they are not added. 

People choose to either scold the author in the replies or add a image descriptions as an answer. 
It would make this process much easier to have a dedicated feature for that.

It would help to emphasize the importance of image descriptions, people that want to write them for others, and of course people that rely on them.

**Comment by RockstarRaccoon** [comment]
Seconded. It gets really awkward to remind people of this in actual replies, because it feels a bit rude to be like, "Hey you forgot the caption!"

**Comment by rkingett** [comment]
I'd really love this to happen as well! The Alt4Me hashtag doesn't carry to all instances, and I'd love something to be built to suggest media descriptions. This way, users that would love to provide them can do so without making the OP feel bad. This would also work far better than groups.

**Comment by Taywee** [comment]
Another benefit of a dedicated feature is that alt text often has higher character limits than the posts themselves.  mastodon.art, for example, has a limit of 500 characters on posts, but 1500 for alt text.  A dedicated feature would be able to respect the appropriate character limit.

**Comment by SteffoSpieler** [comment]
This would be a great feature, but I think the OP of a post should get a message like "xyz suggested an alt text, do you want to apply it?" first, so trolls can't do a lot of harm by writing bad stuff into the alt texts.
Also this warning should obviously not appear if it's from someone who OP has blocked or similar.

**Comment by OddDev** [comment]
> This would be a great feature, but I think the OP of a post should get a message like "xyz suggested an alt text, do you want to apply it?" first, so trolls can't do a lot of harm by writing bad stuff into the alt texts.
> Also this warning should obviously not appear if it's from someone who OP has blocked or similar.

That's what I was suggesting 😅

> One can add a suitable description for the author to approve.

**Comment by SteffoSpieler** [comment]
> That's what I was suggesting 😅

oh whoops, then I've missed that. Sorry 😅

**Comment by sandyarmstrong** [comment]
> People choose to either scold the author in the replies

Yup, this is a huge problem. A supported way to propose alt text without spamming replies would do a lot for Mastodon communities.

I could see this UX being "just" a special kind of reply that isn't shown with other replies.

**Comment by Diwannist** [comment]
Some images on Mastodon might not have alt text.

Allowing users other than the original poster to suggest alt text—and making it easy for the original poster to accept the suggestion—might increase the ratio of images with alt text relative to images without alt text.
